### ISSUE TITLE
Request: "Fill triangle" block in Pen extension

### ISSUE THREAD

**Issue by towerofnix** [LLK/scratch-gui#1451]
Now that the Pen category has been split into its own extension, there's a bit more leeway to add more "advanced" blocks to it. I suggest a "fill triangle" block, which simply fills a triangle with the current pen color/opacity over the area specified by the given coordinates.

Two reasons for the "fill triangle" block:

1. It would be *fast*. There are ways to fill a triangle in Scratch already, but these require manually putting together a Scratch script and drawing lots of lines one by one, which is quite costly on performance. Having a dedicated block would speed up any projects which use filled pen shapes greatly, and would make other projects that were previously out of the question because of pen performance be more realistic.

2. It would be *basic* - in the sense of the word "base". Having a "fill triangle" block would prompt users to make their own "fill square", "fill quadrilateral", etc blocks. And these would all be relatively simple to implement, since quadrilaterals (and other polygons) can be split into triangles. Building these custom blocks would clearly and practically show to users how useful custom blocks can be.

**Comment by mrjacobbloom** [comment]
crosslinking with LLK/scratch-flash#1143

**Comment by griffpatch** [comment]
he he - I still can't believe I had the time to implement a scratch emulator back then, but it was fun being able to demo a filled polygon block that way ;) - https://scratch.mit.edu/projects/113321949/ - However, a filled triangle is a very good start... The only issue might be when trying to draw semi transparent polygons as some of seams would have to be diagonal and the aliasing would not work between semi transparent edges where the colour was supposed to be uniform. But hey ;) - I'd still prefer that over not having a filled polygon of any type!

**Comment by hedgehog125** [comment]
Yes! That would be really good, maybe similar to my suggestion I wrote a little while ago: https://scratch.mit.edu/discuss/topic/288708/
